src/core/net/nd6.hpp

```diff
@@ -525,6 +564,53 @@ class RaFlagsExtOption : public Option, private Clearable<RaFlagsExtOption>
 
 static_assert(sizeof(RaFlagsExtOption) == 8, "invalid RaFlagsExtOption structure");
 
+
+class ND6TxMessage {
+
+    public:
+        /**
+         * Gets the prepared ND6 message as an `Icmp6Packet`.
+         *
+         * @param[out] aPacket   A reference to an `Icmp6Packet`.
+         *
+         */
+        void GetAsPacket(Icmp6Packet &aPacket) const { aPacket.Init(mArray.AsCArray(), mArray.GetLength()); }
+
+        /**
+         * Appends bytes from a given buffer to the ND6 message.
+         *
+         * @param[in] aBytes     A pointer to the buffer containing the bytes to append.
+         * @param[in] aLength    The buffer length.
+         *
+         * @retval kErrorNone    Bytes are appended successfully.
+         * @retval kErrorNoBufs  Insufficient available buffers to grow the message.
+         *
+         */
+        Error AppendBytes(const uint8_t *aBytes, uint16_t aLength);
+
```

Comment:
Can add the following template method, then we can append any fixed size object, e.g., `Header` types?

```c++
    /**
     * Appends an object to the ND6 message.
     *
     * @tparam    ObjectType   The object type to append to the message.
     *
     * @param[in] aObject      A reference to the object to append to the message.
     *
     * @retval kErrorNone      Successfully appended the object.
     * @retval kErrorNoBufs  Insufficient available buffers to grow the message.
     *
     */
    template <typename ObjectType> Error Append(const ObjectType &aObject)
    {
        static_assert(!TypeTraits::IsPointer<ObjectType>::kValue, "ObjectType must not be a pointer");

        return AppendBytes(reintepret_cast<const uint8_t>(&aObject), sizeof(ObjectType));
    }
```

With this we can replace 
```nsMsg.AppendBytes(reinterpret_cast<const uint8_t *>(&nsHdr), sizeof(nsHdr));``` 

to 
```nsMsg.Append(ndHeader);```